packages/medusa/src/api/routes/store/products/index.ts

```diff
@@ -97,6 +97,7 @@ export const allowedStoreProductsRelations = [
   "variants.title",
   "variants.prices.amount",
   "sales_channels",
+  "categories",
```

Comment:
thought: Should we guard the Product Categories Store API with a feature flag similar to the Admin API?